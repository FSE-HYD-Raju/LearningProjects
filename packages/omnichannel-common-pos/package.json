{
	"name": "omnichannel-common-pos",
	"version": "111.2.0",
	"description": "Common actions, stores and components for omnichannel",
	"main": "lib/index",
	"types": "lib/index.d.ts",
	"oc:es6": "src/index",
	"scripts": {
		"test": "../../testSinglePackage.sh $(pwd)",
		"test-watch": "jest --watch",
		"lint": "eslint tests src --max-warnings=0 ",
		"test-ci": "jest",
		"lint-fix": "eslint src tests --fix",
		"flow": "flow; test $? -eq 0 -o $? -eq 2",
		"clean": "rm -rf lib",
		"build-ts": "tsc --pretty",
		"copy-assets": "copyfiles --up 1 'src/**/*.@(scss|jpg|jpeg|gif|png|svg)' ./lib",
		"build": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -s inline -d lib/ src/ && npm run build-ts && npm run copy-assets && rimraf -rf messages",
		"watch": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -w -s inline -d lib/ src/ && rimraf -rf messages",
		"prepublish": "npm run clean && npm run build"
	},
	"jest": {
		"testRegex": "(/tests/.*)\\.test\\.js$",
		"coverageDirectory": "jestcoverage",
		"coverageReporters": [
			"lcov",
			"html"
		],
		"setupFiles": [
			"../../jest-setup.js"
		]
	},
	"author": "",
	"license": "ISC",
	"publishConfig": {
		"registry": "https://artifactory.qvantel.net/artifactory/api/npm/npm-local"
	},
	"dependencies": {
		"alt": "0.18.6",
		"alt-utils": "2.0.0",
		"autosuggest-highlight": "2.1.1",
		"axios": "0.18.0",
		"babel-cli": "6.26.0",
		"babel-core": "6.26.0",
		"babel-eslint": "7.2.3",
		"babel-jest": "23.6.0",
		"babel-loader": "7.1.3",
		"babel-plugin-add-module-exports": "0.2.1",
		"babel-plugin-module-resolver": "3.1.0",
		"babel-plugin-react-intl": "2.4.0",
		"babel-plugin-react-transform": "2.0.2",
		"babel-plugin-transform-class-properties": "6.24.1",
		"babel-plugin-transform-decorators-legacy": "1.3.4",
		"babel-plugin-transform-es2015-classes": "6.24.1",
		"babel-plugin-transform-proto-to-assign": "6.26.0",
		"babel-plugin-transform-react-constant-elements": "6.23.0",
		"babel-plugin-transform-react-inline-elements": "6.22.0",
		"babel-plugin-transform-react-pure-class-to-function": "1.0.1",
		"babel-plugin-transform-react-remove-prop-types": "0.4.20",
		"babel-plugin-transform-runtime": "6.23.0",
		"babel-plugin-typecheck": "3.9.0",
		"babel-polyfill": "6.26.0",
		"babel-preset-es2015": "6.24.1",
		"babel-preset-es2015-loose": "7.0.0",
		"babel-preset-react": "6.24.1",
		"babel-preset-stage-0": "6.24.1",
		"babel-register": "6.26.0",
		"babel-runtime": "6.26.0",
		"bootstrap-sass": "3.3.7",
		"byte-size": "4.0.4",
		"classnames": "2.2.6",
		"connected-react-router": "4.3.0",
		"copyfiles": "2.0.0",
		"country-data": "0.0.31",
		"cross-env": "5.1.4",
		"crypto-js": "3.1.9-1",
		"css-ns": "1.2.1",
		"debug": "4.1.0",
		"diff": "3.4.0",
		"history": "4.7.2",
		"jwt-decode": "2.2.0",
		"langs": "2.0.0",
		"lodash": "4.17.11",
		"moment": "2.21.0",
		"moment-duration-format": "2.2.2",
		"object-assign": "4.1.1",
		"omnichannel-flow-pos": "111.2.0",
		"path": "0.12.7",
		"path-to-regexp": "2.3.0",
		"prop-types": "15.6.1",
		"ramda": "0.24.1",
		"rc-progress": "2.2.5",
		"rc-slider": "8.6.1",
		"react": "16.10.2",
		"react-autosuggest": "9.3.4",
		"react-bootstrap": "0.32.1",
		"react-collapse": "4.0.3",
		"react-confirm": "0.1.14",
		"react-dates": "20.1.0",
		"react-dom": "16.10.2",
		"react-dropzone": "3.13.4",
		"react-flatpickr": "3.6.4",
		"react-formal": "0.25.4",
		"react-formal-inputs": "2.1.1",
		"react-helmet": "5.2.0",
		"react-highlight-words": "0.11.0",
		"react-input-range": "1.3.0",
		"react-intl": "2.7.0",
		"react-motion": "0.5.2",
		"react-motion-ui-pack": "0.10.3",
		"react-onclickoutside": "6.7.1",
		"react-redux": "5.0.7",
		"react-router": "4.2.0",
		"react-router-dom": "4.2.2",
		"react-select": "1.2.1",
		"react-toastr": "3.0.0",
		"react-widgets": "3.5.0",
		"recompose": "0.30.0",
		"redux": "4.0.0",
		"redux-saga": "0.16.0",
		"redux-saga-tester": "1.0.379",
		"shallowequal": "0.2.2",
		"twix": "1.2.1",
		"universal-cookie": "2.1.2",
		"uuid": "3.2.1",
		"weasel-styles": "2.3.1",
		"yup": "0.21.3"
	},
	"devDependencies": {
		"@types/alt": "0.16.35",
		"@types/classnames": "2.2.6",
		"@types/jest": "23.3.9",
		"@types/lodash": "4.14.118",
		"@types/moment-duration-format": "2.2.2",
		"@types/node": "8.0.0",
		"@types/rc-slider": "8.2.3",
		"@types/react": "16.7.18",
		"@types/react-bootstrap": "0.32.14",
		"@types/react-collapse": "4.0.2",
		"@types/react-confirm": "0.1.2",
		"@types/react-dates": "17.1.5",
		"@types/react-dom": "16.0.5",
		"@types/react-flatpickr": "3.7.0",
		"@types/react-helmet": "5.0.7",
		"@types/react-highlight-words": "0.12.0",
		"@types/react-intl": "2.3.9",
		"@types/react-motion": "0.0.27",
		"@types/react-motion-ui-pack": "0.10.1",
		"@types/react-onclickoutside": "6.0.4",
		"@types/react-redux": "6.0.0",
		"@types/react-router": "4.0.25",
		"@types/react-select": "1.2.0",
		"@types/react-toastr": "3.0.3",
		"@types/recompose": "0.26.4",
		"@types/redux-actions": "2.3.0",
		"@types/redux-devtools": "3.0.44",
		"@types/redux-devtools-dock-monitor": "1.1.32",
		"@types/redux-devtools-log-monitor": "1.0.33",
		"@types/redux-logger": "3.0.6",
		"@types/seamless-immutable": "7.1.2",
		"@types/universal-cookie": "2.2.0",
		"@types/webpack": "4.4.0",
		"@types/webpack-env": "1.13.6",
		"@types/yup": "0.26.0",
		"babel-plugin-__coverage__": "11.0.0",
		"babel-plugin-transform-flow-strip-types": "6.22.0",
		"babel-plugin-transform-runtime": "6.23.0",
		"enzyme": "3.7.0",
		"enzyme-adapter-react-16": "1.10.0",
		"enzyme-to-json": "3.3.3",
		"eslint": "4.10.0",
		"eslint-config-omnichannel": "2.2.4",
		"eslint-config-prettier": "1.7.0",
		"eslint-config-react-app": "2.1.0",
		"eslint-loader": "1.9.0",
		"eslint-plugin-babel": "4.1.2",
		"eslint-plugin-flowtype": "2.39.1",
		"eslint-plugin-import": "2.8.0",
		"eslint-plugin-jsx-a11y": "5.1.1",
		"eslint-plugin-prettier": "2.6.0",
		"eslint-plugin-react": "7.4.0",
		"eslint-plugin-react-intl": "1.1.1",
		"flow-bin": "0.70.0",
		"jest": "23.6.0",
		"jest-junit": "3.6.0",
		"lint-staged": "4.3.0",
		"moxios": "0.4.0",
		"pre-commit": "1.2.2",
		"prettier": "1.6.1",
		"react-router-enzyme-context": "1.2.0",
		"react-test-renderer": "16.7.0",
		"redux-devtools": "3.4.1",
		"redux-devtools-dock-monitor": "1.1.3",
		"redux-devtools-log-monitor": "1.4.0",
		"redux-logger": "3.0.6",
		"rimraf": "2.6.2",
		"sinon": "4.3.0",
		"ts-jest": "23.10.5",
		"ts-loader": "5.3.1",
		"typescript": "3.1.6",
		"webpack": "4.26.1"
	},
	"commitInfo": "omni-ui-common-pos|184f176|1571059113|Venkata_Gangadhar_Kodela|develop"
}
