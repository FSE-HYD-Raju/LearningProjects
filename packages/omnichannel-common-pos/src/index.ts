export * from "./components/ocComponents";
import isCheckboxRendering from "./components/product/configuration/groups/isProductOfferingGroupCheckboxRendering";
import ActivateSimCardContainer from "./components/activateSim/ActivateSimCardContainer";
import activateSimMessages from "./components/activateSim/activateSim.messages";
export * from "./components/activateSim/ActivateSimModal";
import SimExampleView from "./components/activateSim/simExampleView/SimExampleView";
import CreditInfoModal from "./components/productLoans/overview/CreditInfoModal";
const historyWithForceRefresh = require("./utils/historyWithForceRefresh");
const history = require("./utils/history");
import apiUrl, { cmsApiUrl } from "./utils/urls";
const Apicalls = require("./utils/Apicalls");
import isClient from "./utils/isClient";
import BasketValidationUtil from "./utils/BasketValidationUtil";
import BasketUtil from "./utils/BasketUtil";
import LocationUtil from "./utils/LocationUtil";
export * from "./components/fluid/FluidContainer";
export * from "./components/subscription";
export * from "./components/login/LoginContainer";
export * from "./components/login/LoginForm";
import CheckableAddon from "./components/checkableAddon/CheckableAddon";
import ChangeLanguage from "./components/NavBar/ChangeLanguage/ChangeLanguage";
import SupportRequestListItem from "./components/supportRequest/SupportRequestListItem";
import SupportRequestDetails from "./components/supportRequest/SupportRequestDetails";
import Langs from "./components/NavBar/ChangeLanguage/Langs";
export * from "./components/error/ErrorModal";

export * from "./components/error/ErrorLabel";
import NavBar from "./components/NavBar/NavBar";
import NavBarContainer from "./components/NavBar/NavBarContainer";
import NavBarButton from "./components/NavBar/NavBarButton/NavBarButton";
import NavBarButtons from "./components/NavBar/NavBarButtons/NavBarButtons";
const AltContainer = require("./components/AltContainer");
export * from "./components/categories/CategoryTiles";
import CategoryTilesContainer from "./components/categories/CategoryTilesContainer";

import AALoginCallback from "./components/auth/AALoginCallback";
import AALoginCallbackContainer from "./components/auth/AALoginCallbackContainer";
import LuhnCheck from "./utils/LuhnCheck";

import ErrorCodes from "./utils/ErrorCodes";
import cssns from "./utils/cssnsConfig";
import PaymentUtil from "./utils/PaymentUtil";
import ProductUtil from "./utils/product/ProductUtil";
import ProductOfferingUtil from "./utils/ProductOfferingUtil";
import CurrencyUtil from "./utils/CurrencyUtil";

import MsisdnConfigurationUtils from "./components/product/msisdn/utils/msisdnConfiguration.utils";
import PosMsisdnConfigurationModal, { PosMsisdnConfigurationModalProps, PosMsisdnConfigurationModalState,
	PosMsisdnConfigurationModalActionProps,
	PosMsisdnConfigurationModalStateProps } from "./components/product/msisdn/PosMsisdnConfigurationModal";
import withProductUtil from "./utils/product/withProductUtil";
import withProductOfferingUtil from "./utils/product/withProductOfferingUtil";
import WindowSize from "./components/windowSize/WindowSize";
import Flex from "./components/flex/Flex";
import PaymentSelection, { PaymentSelectionProps } from "./components/payment/PaymentSelection";
import PaymentReturnHandler from "./components/payment/PaymentReturnHandler";
import PaymentReturnHandlerContainer, {
	PaymentReturnHandlerContainerOwnProps
} from "./components/payment/PaymentReturnHandlerContainer";
import PaymentMethodsReturnHandler from "./components/paymentMethods/PaymentMethodsReturnHandler";
import PaymentMethodsReturnHandlerContainer from "./components/paymentMethods/PaymentMethodsReturnHandlerContainer";
import MiniBasket from "./components/basket/minibasket/MiniBasket";
import MiniBasketContainer from "./components/basket/minibasket/MiniBasketContainer";
import BasketDiscountContainer from "./components/basket/discount/BasketDiscountContainer";
import BasketDiscountContainerRedux from "./components/basket/discount/BasketDiscountContainerRedux";
import SalesBasket from "./components/basket/sales/SalesBasket";
import SalesBasketContainer from "./components/basket/sales/SalesBasketContainer";
import ToasterContainer from "./components/toaster/ToasterContainer";

import CharacteristicsComparisonModal from "./components/shop/comparison/CharacteristicsComparisonModal";
import CharacteristicsComparisonModalContainer from "./components/shop/comparison/CharacteristicsComparisonModalContainer";
import ComparisonFooter from "./components/shop/comparison/ComparisonFooter";
import ComparisonFooterContainer from "./components/shop/comparison/ComparisonFooterContainer";
import ComparisonPrice from "./components/shop/comparison/ComparisonPrice";
import ShopProductImage from "./components/shop/product/ShopProductImage";
import ShopBundleImage from "./components/shop/product/ShopBundleImage";
import ProductServicesByActivityContainer from "./components/subscription/services/ProductServicesByActivityContainer";
import ActiveAgreementSelect from "./components/basket/ActiveAgreementSelect";
import ActiveAgreementSelectContainer from "./components/basket/ActiveAgreementSelectContainer";
import MiniBasketErrorContainer from "./components/basket/minibasket/MiniBasketErrorContainer";
import AddressValidationHandler from "./components/addressValidationHandler/AddressValidationHandler";
import AddressValidationHandlerContainer from "./components/addressValidationHandler/AddressValidationHandlerContainer";
import LocationSelectionView from "./components/locationSelection/LocationSelectionView";
import VersionInformationContainer from "./components/versionInformation/VersionInformationContainer";
export * from "./components/mockWrapper/MockWrapper";
import CustomerListDropdownContainer from "./components/NavBar/CustomerListDropdown/CustomerListDropdownContainer";
import ProductOfferingConfiguration from "./components/product/ProductOfferingConfiguration";
import ProductOfferingConfigurationContainer from "./components/product/ProductOfferingConfigurationContainer";
export * from "./components/product/withProductConfigurations";
export * from "./components/product/ProductPriceType";
import FilterControls from "./components/product/filters/FilterControls";
import FilterControlsContainer from "./components/product/filters/FilterControlsContainer";
import CheckboxListProductFilter from "./components/filters/CheckboxListProductFilter";
import CharacteristicsDropdown from "./components/product/characteristics/CharacteristicsDropdown";
import CharacteristicsTextInputContainer from "./components/product/characteristics/CharacteristicsTextInputContainer";
import CharacteristicsDate from "./components/product/characteristics/CharacteristicsDate";
export * from "./components/product/characteristics/InputTypes";
import PlanUtils from "./components/subscription/plan/Plan.utils";
import planMessages from "./components/subscription/plan/Plans.messages";
export * from "./components/subscription/plan/list/PlanRow";
import AddonPrice, { AddonPriceProps } from "./components/subscription/addon/AddonPrice";
import SessionStatePollingIFrameContainer from "./components/auth/SessionStatePollingIFrameContainer";
import UnitOfMeasure from "./components/unitOfMeasure/UnitOfMeasure";
import unitOfMeasureMessages from "./components/unitOfMeasure/intl/UnitOfMeasure.messages";
import unitOfMeasureShortMessages from "./components/unitOfMeasure/intl/UnitOfMeasureShort.messages";
export * from "./utils/UnitOfMeasureUtil";
export * from "./components/ProductOfferingConfigurationModal";
export * from "./components/subscription/utils/Subscription.utils";
export * from "./components/subscription/utils/ConsumptionUtil";
export * from "./components/subscription/utils/testData/agreement";
import ProductOfferingConfigurationModalContainer from "./components/ProductOfferingConfigurationModalContainer";
export * from "./components/navigation/NavigationConfirmModal";
import OrderUtil from "./utils/order/OrderUtil";
import BalanceError from "./components/payment/BalanceError";
import PaymentFormContainer from "./components/payment/PaymentFormContainer";
import LanguagesUtil from "./utils/LanguagesUtil";
import SimCardUtil from "./utils/simCard/SimCardUtil";
import ConfigurationsUtil from "./utils/ConfigurationsUtil";
import MessageBox from "./components/messageBox/MessageBox";
import BasketDiscountModal from "./components/basket/BasketDiscountModal";
import { BasketSelectionActionEnum } from "./redux/types/BasketSelectionAction";
import MarketingConsentForm from "./components/marketingConsent/MarketingConsentForm";
import RefreshButton from "./components/basket/RefreshButton";

import StringUtil from "./utils/StringUtil";
import UrlUtil from "./utils/UrlUtil";

import extractQueryParamsFromUrl from "./components/auth/extractQueryParamsFromUrl";
import PriceHolder from "./components/prices/PriceHolder";
import UsageDonut, { UsageDonutBaseline, UsageDonutProps } from "./components/UsageDonut/UsageDonut";

import RecurringTopUpConfigurationForm from "./components/topUps/RecurringTopUpConfigurationForm";
import RecurringTopUpConfigurationFormContainer from "./components/topUps/RecurringTopUpConfigurationFormContainer";
import NavBarSearchMenu from "./components/NavBar/NavBarSearchMenu/NavBarSearchMenu";
import NavBarMobileContainer from "./components/NavBar/NavBarMobile/NavBarMobileContainer";
import NavBarMobile from "./components/NavBar/NavBarMobile/NavBarMobile";
import { NavBarMobilePage, digitalLifePages } from "./components/NavBar/NavBarMobile/NavBarMobilePages";
import * as ChannelUtils from "./channelUtils";
import { PortInData } from "./components/product/msisdn/PosMsisdnConfigurationModalContainer";
export * from "./channelUtils";
const Flux = require("./flux");
import * as Cms from "./components/cms";
import CmsContentSpot from "./components/cms/CmsContentSpot";
import CmsPopoverContainer from "./components/cms/popover/CmsPopoverContainer";
export * from "./components/cms/popover/CmsPopoverPreLoadingBase";
import { SessionUtils, SessionKeys } from "./utils/SessionUtils";
import { ContextType, contextTypesValidationMap } from "./types";
import ConfigureModalProvider from "./components/subscription/addon/configuration/ConfigureModalProvider";
import ConfigureModalProviderContainer from "./components/subscription/addon/configuration/ConfigureModalProviderContainer";
import commonMessages from "./commonMessages";
export * from "./components/product/msisdn/withMsisdnConfiguration";
export * from "./customization";
export * from "./components/NavBar/ChangeLanguage/Langs";
export { NavBarStateProps, NavBarProps, NavBarUnwrapped } from "./components/NavBar/NavBar";
export * from  "./channelUtils/UnitOfTime.messages";
export {
	NavBarButtonsActionProps,
	NavBarButtonWithClickOutside,
	NavBarButtonsStateProps
} from "./components/NavBar/NavBarButtons/NavBarButtons";
import NavBarMessages from "./components/NavBar/NavBarButtons/NavBarButtons.messages";

export * from "./selectors";
export * from "./components/route";
export * from "./customization";
import ProductOfferingConfigurationUtil from "./components/product/configuration/utils/ProductOfferingConfigurationUtil";
import ConfigurableSubscriptionUtils from "./components/product/configuration/groups/configurablesubscription/ConfigurableSubscription.utils";
import PostalAddressUtil from "./utils/user/PostalAddressUtil";
import MsisdnUtil from "./utils/MsisdnUtil";
import GridPaginationUtil from "./utils/agGrid/GridPaginationUtil";
import countryMessages from "./country.messages";
import nationalityMessages from "./nationality.messages";
import DeliveryAddressType from "./components/delivery/DeliveryAddressType";
import ChangeSimModalContainer from "./components/changeSim/ChangeSimModalContainer";
export * from "./components/subscription/plan/changePlan/suggestion/ChangePlanSuggestionContainer";
import DeliveryAddressSelection from "./components/delivery/DeliveryAddressSelection";
export * from "./components/delivery/DeliveryAddressSelection";
import DeliveryAddressSelectionContainer from "./components/delivery/DeliveryAddressSelectionContainer";
import NewSupportCaseModalContainer from "./components/supportRequest/NewSupportCaseModalContainer";
import { CreditCardSelectionList } from "./components/payment/creditcardselection/CreditCardSelectionList";
import MsisdnPatternSearch from "./components/product/msisdn/MsisdnPatternSearch";
import PersonDetailsPostalAddressForm from "./components/delivery/forms/PersonDetailsPostalAddressForm";
import AllowanceUtil from "./utils/AllowanceUtil";
import ActivateProductLoanModalContainer from "./components/productLoans/purchase/ActivateProductLoanModalContainer";
import UsageCountersUtil from "./utils/product/UsageCountersUtil";
import ErrorModalContainer from "./components/error/ErrorModalContainer";
import { ErrorMessagesType } from "./components/error/Error.messages";
import SubscriptionCreditInfoContainer from "./components/productLoans/overview/SubscriptionCreditInfoContainer";
import HasActiveLoanTopUpBannerContainer from "./components/productLoans/topUp/HasActiveLoanTopUpBannerContainer";
import {
	default as ServiceStateChangeModalContainer,
	ServiceStateChangeModalContainerBaseline,
} from "./components/subscription/services/ServiceStateChangeModalContainer";
import {
	default as AddonStateChangeModalContainer,
	AddonStateChangeModalContainerBaseline
} from "./components/subscription/addon/AddonStateChangeModalContainer";
import PriceUtil from "./utils/PriceUtil";
import { LOCATIONS_ROOT_LEVEL_QUERY, CountryOptions } from "./redux/types";

import MsisdnSelection from "./components/msisdnSelection/MsisdnSelection";
import MsisdnSelectionContainer from "./components/msisdnSelection/MsisdnSelectionContainer";
import PriceComponent, { PriceComponentWithProductConfiguration } from "./components/priceComponent/PriceComponent";
import ComplexPriceComponent from "./components/priceComponent/ComplexPriceComponent";
import ProductPriceRenderer from "./components/priceComponent/renderers/ProductPriceRenderer";
import StandardPriceRenderer from "./components/priceComponent/renderers/StandardPriceRenderer";
import GridPriceRenderer from "./components/priceComponent/renderers/GridPriceRenderer";

import AddToBasketButtonContainer from "./components/addToBasketButton/AddToBasketButtonContainer";
import ProductAddToBasketButtonContainer from "./components/addToBasketButton/ProductAddToBasketButtonContainer";
import messages from "./components/addToBasketButton/AddToBasketButton.messages";
export * from "./components/loading";
export * from "./components/templatedAnchor/TemplatedAnchor";

export * from "./redux";
export * from "./redux/utils/channel";
export * from "./redux/models/cms/cms.utils";
export * from "./testUtils";
export * from "./schemas";
export * from "./routes/commonRoutesMap";

import * as ConsulUtils from "./redux/models/consul/consul.utils";
import { RecurringTopUpUtil } from "./components/topUps/RecurringTopUpUtil";
import MsisdnSelectionPicker from "./components/msisdnSelection/MsisdnSelectionPicker";
import { AddonsTabView } from "./components/subscription/AddonsTabView";
import SubscriptionState from "./components/subscription/SubscriptionState";
import TranslatedComponent from "./components/TranslatedComponent/TranslatedComponent";
export * from "./components/payment/PaymentByBalanceDetails";
import PaymentDetails from "./components/payment/PaymentDetails";
import NavBarCurrencySelectionContainer from "./components/NavBar/NavBarCurrencySelection/NavBarCurrencySelectionContainer";
import NavBarSessionButtonContainer from "./components/NavBar/NavBarSessionButton/NavBarSessionButtonContainer";
import TopUpContainer from "./components/NavBar/topUp/TopUpContainer";
import BasketMenuContainer from "./components/basket/BasketMenuContainer";
import ChangeDeliveryAddressContainerECare from "./components/delivery/ChangeDeliveryAddressContainerECare";
import RecurringTopUpsContainer from "./components/topUps/RecurringTopUpsContainer";
import RecurringTopUpAddModal from "./components/topUps/RecurringTopUpAddModal";
import RecurringTopUpEditModal from "./components/topUps/RecurringTopUpEditModal";
import AgreementUtil from "./utils/AgreementUtil";
import ArrayUtil from "./utils/ArrayUtil";
import { SubscriptionPlans } from "./components/subscription/plan/SubscriptionPlans";
import { SubscriptionPlansContainer } from "./components/subscription/plan/SubscriptionPlansContainer";
import DateUtil from "./utils/DateUtil";
import CustomerBasketsContainer from "./components/basket/customerBasket/CustomerBasketsContainer";

export * from "./components/subscription/AddonsTabView";
export * from "./components/subscription/addon/accordion/AddonRow";
export * from "./components/subscription/AddonsTabLifecycleFilter";
export * from "./components/subscription/addon/AddonsViewContent";
export * from "./components/EmptyComponent";
export * from "./components/topUps/forms/RecurringTopUpAmount";
export {
	ArrayUtil,
	AgreementUtil,
	SubscriptionPlans,
	SubscriptionPlansContainer,
	MsisdnSelectionPicker,
	Flux,
	ChannelUtils,
	Cms,
	CmsContentSpot,
	CmsPopoverContainer,
	apiUrl,
	cmsApiUrl,
	historyWithForceRefresh,
	history,
	Apicalls,
	isClient,
	BasketUtil,
	BasketValidationUtil,
	BasketSelectionActionEnum,
	LocationUtil,
	CheckableAddon,
	ChangeLanguage,
	Langs,
	ProductOfferingConfigurationUtil,
	ConfigurableSubscriptionUtils,
	ErrorModalContainer,
	ErrorMessagesType,
	BasketMenuContainer,
	TopUpContainer,
	NavBar,
	NavBarButton,
	NavBarButtons,
	NavBarMessages,
	NavBarContainer,
	NavBarMobilePage,
	digitalLifePages,
	NavBarCurrencySelectionContainer,
	NavBarSessionButtonContainer,
	contextTypesValidationMap,
	AltContainer,
	CategoryTilesContainer,
	AALoginCallback,
	AALoginCallbackContainer,
	NewSupportCaseModalContainer,
	LuhnCheck,
	ErrorCodes,
	cssns,
	ProductUtil,
	ProductOfferingUtil,
	PriceUtil,
	AllowanceUtil,
	UsageCountersUtil,
	PostalAddressUtil,
	MsisdnUtil,
	GridPaginationUtil,
	MsisdnConfigurationUtils,
	withProductUtil,
	withProductOfferingUtil,
	WindowSize,
	Flex,
	PaymentSelection,
	PaymentSelectionProps,
	PaymentReturnHandler,
	PaymentReturnHandlerContainer,
	PaymentReturnHandlerContainerOwnProps,
	PaymentMethodsReturnHandler,
	PaymentMethodsReturnHandlerContainer,
	isCheckboxRendering,
	ToasterContainer,
	ServiceStateChangeModalContainer,
	ProductServicesByActivityContainer,
	CharacteristicsComparisonModal,
	CharacteristicsComparisonModalContainer,
	ComparisonFooter,
	ComparisonFooterContainer,
	ComparisonPrice,
	ShopProductImage,
	ShopBundleImage,
	SupportRequestListItem,
	SupportRequestDetails,
	ServiceStateChangeModalContainerBaseline,
	AddonStateChangeModalContainer,
	AddonStateChangeModalContainerBaseline,
	ActiveAgreementSelect,
	ActiveAgreementSelectContainer,
	AddressValidationHandler,
	AddressValidationHandlerContainer,
	LocationSelectionView,
	VersionInformationContainer,
	CustomerListDropdownContainer,
	ProductOfferingConfiguration,
	ProductOfferingConfigurationContainer,
	FilterControls,
	FilterControlsContainer,
	CheckboxListProductFilter,
	PlanUtils,
	AddonPrice,
	AddonPriceProps,
	SessionStatePollingIFrameContainer,
	SubscriptionState,
	ProductOfferingConfigurationModalContainer,
	OrderUtil,
	BalanceError,
	PaymentFormContainer,
	LanguagesUtil,
	SimCardUtil,
	ConfigurationsUtil,
	MessageBox,
	BasketDiscountModal,
	MarketingConsentForm,
	RefreshButton,
	SessionKeys,
	SessionUtils,
	StringUtil,
	extractQueryParamsFromUrl,
	PriceHolder,
	UsageDonut,
	UsageDonutBaseline,
	UsageDonutProps,
	RecurringTopUpConfigurationForm,
	RecurringTopUpConfigurationFormContainer,
	TranslatedComponent,
	ContextType,
	ConfigureModalProvider,
	ConfigureModalProviderContainer,
	NavBarSearchMenu,
	NavBarMobileContainer,
	NavBarMobile,
	PaymentUtil,
	UnitOfMeasure,
	unitOfMeasureMessages,
	unitOfMeasureShortMessages,
	commonMessages,
	countryMessages,
	nationalityMessages,
	CountryOptions,
	DeliveryAddressSelectionContainer,
	DeliveryAddressType,
	ChangeSimModalContainer,
	DeliveryAddressSelection,
	CreditCardSelectionList,
	ActivateSimCardContainer,
	activateSimMessages,
	SimExampleView,
	MsisdnPatternSearch,
	PersonDetailsPostalAddressForm,
	ActivateProductLoanModalContainer,
	CreditInfoModal,
	SubscriptionCreditInfoContainer,
	HasActiveLoanTopUpBannerContainer,
	MsisdnSelection,
	MsisdnSelectionContainer,
	CharacteristicsDropdown,
	CharacteristicsDate,
	CharacteristicsTextInputContainer,
	AddonsTabView,
	MiniBasket,
	MiniBasketContainer,
	BasketDiscountContainer,
	BasketDiscountContainerRedux,
	SalesBasket,
	SalesBasketContainer,
	PaymentDetails,
	LOCATIONS_ROOT_LEVEL_QUERY,
	planMessages,
	PriceComponent,
	PriceComponentWithProductConfiguration,
	ComplexPriceComponent,
	ProductPriceRenderer,
	StandardPriceRenderer,
	GridPriceRenderer,
	AddToBasketButtonContainer,
	ProductAddToBasketButtonContainer,
	ChangeDeliveryAddressContainerECare,
	messages,
	ConsulUtils,
	MiniBasketErrorContainer,
	RecurringTopUpsContainer,
	RecurringTopUpAddModal,
	RecurringTopUpEditModal,
	RecurringTopUpUtil,
	CurrencyUtil,
	PosMsisdnConfigurationModal,
	PosMsisdnConfigurationModalProps,
	PosMsisdnConfigurationModalState,
	PosMsisdnConfigurationModalActionProps,
	PosMsisdnConfigurationModalStateProps,
	PortInData,
	UrlUtil,
	DateUtil,
	CustomerBasketsContainer,
};
