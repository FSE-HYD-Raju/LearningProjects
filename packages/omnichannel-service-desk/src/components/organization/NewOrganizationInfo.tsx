"use strict";

import * as React from "react";
import { Component } from "react";
import { FormattedMessage, OcButton, OcButtonType } from "omnichannel-common-pos";
import messages from "./Organization.messages";

interface NewOrganizationInfoProps {
	actions: {
		clearOrganizationState?: () => void;
		showCustomerCreationModal?: (show: boolean) => void;
	};
}

class NewOrganizationInfo extends Component<NewOrganizationInfoProps> {

	componentWillUnmount() {
		if (this.props.actions.clearOrganizationState) {
			this.props.actions.clearOrganizationState();
		}
	}

	handleClose = () => {
		if (this.props.actions.clearOrganizationState) {
			this.props.actions.clearOrganizationState();
		}
		if (this.props.actions.showCustomerCreationModal) {
			this.props.actions.showCustomerCreationModal(false);
		}
	};

	render() {
		return (
			<div className="organization-creation-success">
				<FormattedMessage {...messages.organizationWasCreatedSuccessfully}/>
				<footer className="customerDataFormFooter modal-footer">
					<OcButton
						id="organization-creation-success-ok-button"
						buttonType={OcButtonType.SUCCESS}
						htmlBtnType="submit"
						onClick={this.handleClose}
					>
						<FormattedMessage {...messages.organizationCustomerCreated}/>
					</OcButton>
				</footer>
			</div>
		);
	}
}

export default NewOrganizationInfo;
export {
	NewOrganizationInfoProps
};
