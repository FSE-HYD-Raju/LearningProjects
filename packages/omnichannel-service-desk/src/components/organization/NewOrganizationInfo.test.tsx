import * as React from "react";
import { shallowWithContext, mountWithContext } from "omnichannel-common-pos";
import NewOrganizationInfo from "./NewOrganizationInfo";

describe("NewOrganizationInfo", () => {
	const minProps = {
		actions: {},
	};

	it("succeeds at shallow mount with minimum props", () => {
		const wrapper = shallowWithContext(<NewOrganizationInfo {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("renders with minimum props", () => {
		const wrapper = mountWithContext(<NewOrganizationInfo {...minProps} />);
		const newOrganizationInfo = wrapper.find(".organization-creation-success");
		expect(newOrganizationInfo.hostNodes()).toHaveLength(1);

		expect(wrapper.find("#organization-creation-success-ok-button").hostNodes()).toHaveLength(1);
	});
});
