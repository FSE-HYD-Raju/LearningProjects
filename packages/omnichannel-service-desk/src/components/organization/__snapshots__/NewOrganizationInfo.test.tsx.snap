// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewOrganizationInfo succeeds at shallow mount with minimum props 1`] = `
<div
  className="organization-creation-success"
>
  <FormattedMessageWrapper
    defaultMessage="Organization was created successfully!"
    description="Create organizational customer success"
    id="organization-creation-form-success"
    wrapperType="wrapper"
  />
  <footer
    className="customerDataFormFooter modal-footer"
  >
    <OcButton
      buttonType="SUCCESS"
      htmlBtnType="submit"
      id="organization-creation-success-ok-button"
      onClick={[Function]}
    >
      <FormattedMessageWrapper
        defaultMessage="OK"
        description="Organization customer created, press OK to proceed"
        id="organization-creation-success-ok-button"
        wrapperType="wrapper"
      />
    </OcButton>
  </footer>
</div>
`;
